File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let mp=new Map();
        for (let val of nums) {
            mp.set(val, (mp.get(val) || 0) + 1);
            if(mp.get(val) >1){
                return true;
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
   public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        set<vector<int>> resp;
        vector<vector<int>> resp2;
        for (int i = 0; i < nums.size(); i++) {
            int l = i+1, r = nums.size() - 1;
            while (l < r) {
                if (l == i) {
                    l++;
                    continue;
                }
                if (r == i) {
                    r--;
                    continue;
                }
                if (nums[l] + nums[r] + nums[i] < 0)
                    l++;
                else if (nums[l] + nums[r] + nums[i] > 0)
                    r--;
                else if (nums[l] + nums[r] + nums[i] == 0) {
                    resp.insert({nums[l], nums[r], nums[i]});
                    l++;
                    r--;
                }
            }
        }
        for (const auto& row : resp) {
            resp2.push_back(row);
        }
        return resp2;
    }
};
